src/resize-sensor/ResizeSensor.js

```diff
@@ -1,51 +1,42 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { props, t, pure } from '../utils';
+import { props, t } from '../utils';
 import _ElementQueries from 'css-element-queries/src/ElementQueries';
 import _ResizeSensor from 'css-element-queries/src/ResizeSensor';
 
-@pure
 @props({
   children: t.ReactChildren,
   onResize: t.Function
 })
 export default class ResizeSensor extends React.Component {
 
   attachResizeSensor = () => {
-    this.resizeSensor = new _ResizeSensor(
-      ReactDOM.findDOMNode(this),
-      this.onResize
-    );
-  }
-
-  detachResizeSensor = () => {
-    this.resizeSensor.detach(ReactDOM.findDOMNode(this));
-  }
-
-  updateResizeSensor = () => {
-    if (this.resizeSensor) {
-      this.detachResizeSensor();
+    const element = ReactDOM.findDOMNode(this);
+    if (!element.resizedAttached) {
+      this.resizeSensor = new _ResizeSensor(element, this.onResize);
     }
-    this.attachResizeSensor();
   }
 
   componentDidMount() {
     _ElementQueries.listen();
-    this.updateResizeSensor();
+    this.attachResizeSensor();
   }
 
   componentDidUpdate() {
-    this.updateResizeSensor();
-  }
-
-  componentWillUnmount() {
```

Comment:
In `componentDidUpdate` this is not true as `findDOMNode` will always be `true` because of  the newly `this.refs.text` already being rendered... (we should check if it contains the resize element added by `ResizeSensor` to understand if there's something to detach or not).

Nevertheless we *should* be able to call it on `componentWillUnmount`, do you think it could be enough adding it in `cwum`?